### PR TITLE
Dialogs now positioned above the game field

### PR DESCRIPTION
**Pull request by TesterNick**

### CODE REVIEW

**Comment by earshinov** [comment]
I have found a few problems:

1. The program hangs if user tries to close it when a dialog window is open:

hang

2. Dialog windows are not positioned properly when using a multi-monitor setup:

multi-monitor setup

3. Dialog windows do not behave exactly like modal windows. To name a few things:
- They appear on Windows task bar
- Clicking the main window while a dialog window is opened causes flickering of the main window title.

flickering

It does not happen with modal windows in other applications (tested in Thunderbird email client, but I am sure you can find another example).

----

I am not sure that all the problems are fixable if using Tcl/Tk, but at least they require investigation.

By the way, the `grab_set` function described at http://effbot.org/tkinterbook/tkinter-dialog-windows.htm might come in handy.

File: app/dialog_window.py [hunk 1]
```diff
@@ -7,12 +7,32 @@ def __init__(self, app):
         self.field = app.field
         self.pause_the_game()
         super().__init__()
+        self.withdraw()
```
Comment (earshinov) [inline app/dialog_window.py]: What is the purpose of this call?

**Review by earshinov — CHANGES_REQUESTED** [review]

**Comment by TesterNick** [comment]
1. Thank you! With https://github.com/TesterNick/Minesweeper/pull/7/commits/aa17379c44fd00757947b9217e5df6885bcaf578 I have fixed multiple windows appearance (at least I could not reproduce the issue anymore).
2. Unfortunately, Ihave no second display anymore. I will create an issue for future investigation. Actually this problem was here before windows positioning.
3. Now dialogs don't appear on taskbar, but I could not fix flickering yet.

File: app/dialog_window.py (hunk 1 above)
Comment (TesterNick) [reply app/dialog_window.py]: The purpose is to hide the window until the program figures out it's geometry.

**Comment by TesterNick** [comment]
I created an issue for windows flickering. I will try to fix it later.

**Pull request merged**